<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1500</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1500</width>
    <height>800</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1500</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1261</width>
      <height>771</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="gridLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>1261</width>
       <height>771</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="2">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>NFA Graph</string>
        </property>
        <widget class="QTableWidget" name="NFA_t">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>621</width>
           <height>361</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>DFA Graph</string>
        </property>
        <widget class="QTableWidget" name="DFA_t">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>621</width>
           <height>361</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>Mini-DFA Graph</string>
        </property>
        <widget class="QTableWidget" name="MINI_t">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>621</width>
           <height>341</height>
          </rect>
         </property>
        </widget>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>Build Code</string>
        </property>
        <widget class="QScrollArea" name="scrollArea">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>20</y>
           <width>621</width>
           <height>351</height>
          </rect>
         </property>
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scrollAreaWidgetContents_2">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>619</width>
            <height>349</height>
           </rect>
          </property>
          <widget class="QTextBrowser" name="Buid_code">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>621</width>
             <height>351</height>
            </rect>
           </property>
          </widget>
         </widget>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>1260</x>
      <y>0</y>
      <width>231</width>
      <height>751</height>
     </rect>
    </property>
    <property name="title">
     <string>Input Expression</string>
    </property>
    <widget class="QScrollArea" name="scrollArea_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>221</width>
       <height>311</height>
      </rect>
     </property>
     <property name="widgetResizable">
      <bool>false</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>219</width>
        <height>249</height>
       </rect>
      </property>
      <widget class="QTextBrowser" name="Input_text">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>221</width>
         <height>551</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
       <property name="html">
        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Please input RE_expression...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="gridLayoutWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>360</y>
       <width>222</width>
       <height>80</height>
      </rect>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="2">
       <widget class="QPushButton" name="clean_t">
        <property name="text">
         <string>clean</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="rFile">
        <property name="text">
         <string>read</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="build_exp">
        <property name="text">
         <string>build</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="sFile">
        <property name="text">
         <string>save</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1500</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionopen_file">
   <property name="text">
    <string>open file</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
